### PR TITLE
Slayer autospecies modification

### PR DESCRIPTION
**Pull request by X0-11**
Moves the slayer GM to the maps folder.
Implements a system to allow species defined in the GM to be automatically applied to players, including custom outfits.
Small fixes to the way special armour interacts with handle_shield

### CODE REVIEW

**Comment by BDpuffy420** [comment]
Requesting review

File: baystation12.dme [hunk 1]
```diff
@@ -2504,6 +2503,8 @@
 #include "maps\insurrection\innie_base_map.dm"
 #include "maps\insurrection\innie_unit_test.dm"
 #include "maps\overmap_example\overmap_example_define.dm"
+#include "maps\slayer\__death_asteroid_map.dm"
```
Comment (bloxgate) [inline baystation12.dme:2506]: What is the reason for moving this?

File: maps/slayer/__death_asteroid_map.dm [hunk 2]
```diff
@@ -5,4 +5,4 @@
 	accessible_z_levels = list("1" = 100)
 	//lobby_icon = 'maps/example/example_lobby.dmi'
 	lobby_icon = 'code/modules/halo/splashworks/title6.png'
-	id_hud_icons = 'code/modules/halo/gamemodes/slayer-team-slayer/slayer_hud_icons.dmi'
+	id_hud_icons = 'maps/slayer/slayer_hud_icons.dmi'
```
Comment (bloxgate) [inline maps/slayer/__death_asteroid_map.dm:8]: Again, what's with the move?

**Review by bloxgate — COMMENTED** [review]
For the most part, LGTM. Just a single question about some stuff being moved.

File: maps/slayer/__death_asteroid_map.dm (hunk 2 above)
Comment (X0-11) [reply maps/slayer/__death_asteroid_map.dm:8]: All other maps have their related files in their /maps folder. The map's movement is so travis runs map-specific tests on it.

**Pull request merged**